amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/coordinator/KinesisClientLibConfiguration.java

```diff
@@ -977,6 +977,9 @@ public KinesisClientLibConfiguration withShardSyncIntervalMillis(long shardSyncI
      */
     public KinesisClientLibConfiguration withMaxRecords(int maxRecords) {
         checkIsValuePositive("MaxRecords", (long) maxRecords);
+        if (maxRecords > DEFAULT_MAX_RECORDS) {
+            throw new IllegalArgumentException("maxRecords must be less than or equal to 10000, but current value is " + maxRecords);
```

Comment:
We should output DEFAULT_MAX_RECORDS instead of 10000. Reason being is in that value were to ever change, this log statement would be incorrect